src/arch/x86/x86_64/src/amdromehsmp.rs

```diff
@@ -0,0 +1,132 @@
+#![allow(non_snake_case)]
```

Comment:
This hiearchy of cpu support in oreboot follows coreboot. 
SOC
CPU
ARCH

the soc in an instance of a cpu; soc has extra IP cores. The SMN code should go in soc I suspect, since it's about a given chip and chiplets.

the CPU is an instance of an ARCH. This file should go there I bet?

We've done a poor job on amd so far of honoring these rules. We did a good job on sifive, so let's look there.

ARCH
arch/riscv/rv32/src/lib.rs arch/riscv/rv32/src/mod.rs arch/riscv/rv32/src/reg.rs arch/riscv/rv64/src/lib.rs arch/riscv/rv64/src/mod.rs arch/riscv/rv64/src/reg.rs
note there are no Cargo.toml here


CPU

In cpu we have as below; here we see some Cargo.toml, but no riscv. There's nothing special about riscv cpus yet? 

cpu/armltd/cortex-a53/Cargo.lock cpu/armltd/cortex-a53/Cargo.toml cpu/armltd/cortex-a53/src cpu/armltd/cortex-a9/Cargo.lock cpu/armltd/cortex-a9/Cargo.toml cpu/armltd/cortex-a9/src cpu/lowrisc/ibex/Cargo.lock cpu/lowrisc/ibex/Cargo.toml cpu/lowrisc/ibex/src

SOC
soc/opentitan/earlgrey/Cargo.lock soc/opentitan/earlgrey/Cargo.toml soc/opentitan/earlgrey/src soc/sifive/fu540/Cargo.lock soc/sifive/fu540/Cargo.toml soc/sifive/fu540/src

for the rest of this patch I'l specular on where things ought to be.

For starters, though, I think this file should be in 
soc/amd/rom/src/hsmp.rs

note that, due to namespacing, we don't need to repeat the path name in the file name. hsmp.rs is fine.